pandas/io/pytables.py

```diff
@@ -2968,11 +2968,17 @@ def write_array_empty(self, key: str, value: ArrayLike):
         node._v_attrs.value_type = str(value.dtype)
         node._v_attrs.shape = value.shape
 
-    def write_array(self, key: str, value: ArrayLike, items: Optional[Index] = None):
-        # TODO: we only have one test that gets here, the only EA
+    def write_array(self, key: str, obj: Series, items: Optional[Index] = None):
+        # TODO: we only have a few tests that get here, the only EA
         #  that gets passed is DatetimeArray, and we never have
         #  both self._filters and EA
-        assert isinstance(value, (np.ndarray, ABCExtensionArray)), type(value)
+
+        if isinstance(obj, (np.ndarray, ABCExtensionArray)):
+            value = obj
+        elif is_datetime64tz_dtype(obj):
+            value = obj.dt._get_values()
```

Comment:
i dont know off the top of my head what _get_values returns, is it a DatetimeArray?  The standard pattern would be to use `extract_array(obj, extract_numpy=True)` at the beginning of this method

Reply:
`_get_values` returns a `DatetimeIndex`

https://github.com/pandas-dev/pandas/blob/10e5ad7109f1bb3c47b48b6f0df185e108c5493d/pandas/core/indexes/accessors.py#L47-54

But actually the `extract_array` pattern does the job of this entire if/elif/else block here - thanks for suggesting! (I'll remember for future)